### PR TITLE
feat: PBI-2 login backend

### PR DESCRIPTION
**Pull request by HugoIr**
### Describe your changes
Implement login endpoint

### Backlog name
[MON-41 Create login backend](https://monapi.atlassian.net/browse/MON-41?atlOrigin=eyJpIjoiMmY5ZGFiZDczYThlNDVhZGEyZDIxOTQ1MDhjMmNhNTkiLCJwIjoiaiJ9)

### Acceptance criteria
- User able to login to their account by using email and password

### Example Request
Screen Shot 2022-09-27 at 14 27 41

### Checklist
- [x] I have performed a self-review of my code
- [x] Code successfully run on local
- [x] Feature / changes tested on local
- [x] No failing unit test
- [ ] Passed UAT Test
- [x] Sonarqube tested 
- [ ] Required any changes to environment variable

### CODE REVIEW

File: login/serializers.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        if not email and not password:
+            response = 'Both email and password are required.'
+            raise serializers.ValidationError({'response': response}, code='authorization')
+        elif not email:
+            response = 'Email is required.'
+            raise serializers.ValidationError({'response': response}, code='authorization')
+        elif not password:
+            response = 'Password is required.'
+            raise serializers.ValidationError({'response': response}, code='authorization')
+
```
Comment (andrew-susanto) [inline login/serializers.py:30]:
```suggestion
class LoginSerializer(serializers.Serializers):
    email = serializers.EmailField()
    password = serializers.CharField()
```
ini gimana kalau pakai serializers biasa aja hug, gaperlu pakai model serializer, dia udah default required true

File: login/views.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        data['token']= generate_token(user)
+        
+        return Response(data=data, status=status.HTTP_200_OK)
+    else:
+        data['response'] = 'Invalid email or password.'
+        return Response(data=data, status=status.HTTP_401_UNAUTHORIZED)            
+    
+def generate_token(user):
+    token, created = Token.objects.get_or_create(user=user)
+    return token.key
```
Comment (andrew-susanto) [inline login/views.py:29]:
```suggestion
def generate_token(user):
    token, created = Token.objects.get_or_create(user=user)
    return token.key
```
ini gimana kalau dibuat di utils.py aja hug, sama pake Token.objects.create aja biar support multiple device

File: login/serializers.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        password = attrs.get('password')
+
+        user = authenticate(request=self.context.get('request'),
+                            username=email, email=email, password=password)
+        if not user:
```
[the remark this answers is not in the document]
Comment (HugoIr) [reply login/serializers.py:39]: okaii

File: login/serializers.py (hunk 1 above)
Comment (HugoIr) [reply login/serializers.py:30]: okii

File: login/views.py (hunk 2 above)
Comment (andrew-susanto) [reply login/views.py:29]: iya hugg, pas usernya logout trus login lagi kasih token baru aja

Comment (andrew-susanto) [reply login/views.py:29]: owhh iyaa, ternyata DRF authtoken ga support multiple token per user

Comment (andrew-susanto) [reply login/views.py:29]: okee2 hugg gapapa kalo gitu get_or_create dulu aja. thank youu

File: login/views.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework.response import Response
+from django.contrib.auth import authenticate
+from rest_framework import status
+
+from login.serializers import LoginSerializer
+from .utils import generate_token
```
Comment (andrew-susanto) [inline login/views.py:7]:
```suggestion
from login.utils import generate_token
```

**Review by andrew-susanto — COMMENTED** [review]
satu lagi hug wkkwk 🙏

File: login/views.py (hunk 4 above)
Comment (HugoIr) [reply login/views.py:7]: oiyaa okee wkkwkwk

**Review by andrew-susanto — APPROVED** [review]

**Review by luthfifahlevi — APPROVED** [review]

**Pull request merged**